common/flatpak-utils.c

```diff
@@ -2711,63 +2711,87 @@ summary_find_refs_list (GVariant   *summary,
   return g_steal_pointer (&refs);
 }
 
-/* This matches all refs from @collection_id that have ref, followed by '.'  as prefix */
-char **
-flatpak_summary_match_subrefs (GVariant   *summary,
-                               const char *collection_id,
-                               const char *ref)
+/* Note: This function relies on the similar structure in both xa.cache and summary refs lists */
```

Comment:
Similar? Or identical? If they’re not identical then I think this comment should give the `GVariantType` for both and make the differences more obvious, otherwise it’s not much use to anyone who has to modify this function in future.

Reply:
In any case, you should give the expected `GVariantType` of `haystack`.

Reply:
similar

Reply:
I don't see how I can give a type when it could be two different types, and someone modifying it should find those types from the source anyway.

Reply:
```
/* Note: This function relies on the similar structure of both `xa.cache` (type: `a{s(tts)}`) and summary refs lists (type: `a(s(taya{sv}))`). */
```